### PR TITLE
User Preferences page cleanup

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Making a few updates and some cleanup for the User Preferences page.

* Pulling the result of if a user may set a timeout into react-query so it's cached
* Wrap Dotfiles and Timeout inputs w/ forms - allows enter key to submit
* Drop in new Button components
* Fix how we update when setting `dotfilesRepo` so we don't mutate state directly

## How to test

* Go to the User Preferences page
* Try setting and clearing your dotfiles repo
* Try setting and clearing a custom timeout

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Review by easyCZ — APPROVED** [review]
LGTM
/hold in case you want another review

File: components/dashboard/src/user-settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/user-settings/Preferences.tsx:122]: **praise**: Thanks for adding the form here! Looking forward to further improve this section here. Cross-posting relevant discussions[[1](https://gitpod.slack.com/archives/C01KPEPQYE6/p1676411374009249)][[2](https://gitpod.slack.com/archives/C01KPEPQYE6/p1676416083599729?thread_ts=1676411374.009249&cid=C01KPEPQYE6)][[3](https://gitpod.slack.com/archives/C01KPEPQYE6/p1676450040320919?thread_ts=1676411374.009249&cid=C01KPEPQYE6)] (internal) about the current interaction and future improvements. Cc @phimae @mikenikles

File: components/dashboard/src/user-settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/user-settings/Preferences.tsx:131]: **praise**: Go button component! 🎊

File: components/dashboard/src/user-settings/Preferences.tsx
Comment (gtsiolis) [inline components/dashboard/src/user-settings/Preferences.tsx:82]: **thought**: Feels so much better we don't have to use an extra onClick function here. 💭

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Review by gtsiolis — APPROVED** [review]
UX LGTM! 🌮 🌮 

Thanks for making these changes, @selfcontained. 🏀 

Approving and removing hold since code changes have been approved in https://github.com/gitpod-io/gitpod/pull/16694#pullrequestreview-1327843100.

/unhold

**Pull request merged**
